cart/application/eventHandling.go

```diff
@@ -47,44 +47,92 @@ func (e *EventReceiver) Notify(ctx context.Context, event flamingo.Event) {
 		}
 	//Handle LoginEvent and Merge Cart
 	case *domain.LoginEvent:
-		if currentEvent == nil {
-			return
-		}
-		if !e.cartReceiverService.ShouldHaveGuestCart(currentEvent.Session) {
-			return
-		}
-		guestCart, err := e.cartReceiverService.ViewGuestCart(ctx, currentEvent.Session)
-		if err != nil {
-			e.logger.WithContext(ctx).Error("LoginEvent - Guestcart cannot be received %v", err)
-			return
-		}
-		if !e.cartReceiverService.IsLoggedIn(ctx, currentEvent.Session) {
-			e.logger.WithContext(ctx).Error("Received LoginEvent but user is not logged in!!!")
-			return
-		}
-		for _, d := range guestCart.Deliveries {
-			e.cartService.UpdateDeliveryInfo(ctx, currentEvent.Session, d.DeliveryInfo.Code, cartDomain.CreateDeliveryInfoUpdateCommand(d.DeliveryInfo))
-			for _, item := range d.Cartitems {
-				e.logger.WithContext(ctx).Debug("Merging item from guest to user cart %v", item)
-				addRequest := e.cartService.BuildAddRequest(ctx, item.MarketplaceCode, item.VariantMarketPlaceCode, item.Qty)
-				e.cartService.AddProduct(ctx, currentEvent.Session, d.DeliveryInfo.Code, addRequest)
+		web.RunWithDetachedContext(ctx, func(ctx context.Context) {
+			if currentEvent == nil {
+				return
 			}
-		}
-
-		if guestCart.HasAppliedCouponCode() {
-			for _, code := range guestCart.AppliedCouponCodes {
-				e.cartService.ApplyVoucher(ctx, currentEvent.Session, code.Code)
+			if !e.cartReceiverService.ShouldHaveGuestCart(currentEvent.Session) {
+				return
+			}
+			guestCart, err := e.cartReceiverService.ViewGuestCart(ctx, currentEvent.Session)
+			if err != nil {
+				e.logger.WithContext(ctx).Error("LoginEvent - GuestCart cannot be received %v", err)
+				return
+			}
+			if !e.cartReceiverService.IsLoggedIn(ctx, currentEvent.Session) {
+				e.logger.WithContext(ctx).Error("Received LoginEvent but user is not logged in!!!")
+				return
+			}
+			customerCart, err := e.cartReceiverService.ViewCart(ctx, currentEvent.Session)
+			if err != nil {
+				e.logger.WithContext(ctx).Error("LoginEvent - customerCart cannot be received %v", err)
+				return
+			}
+			err = e.cartService.DeleteSavedSessionGuestCartID(currentEvent.Session)
```

Comment:
In the old implementation the session was deleted at the end, should we delete it at this point already?

Reply:
That was by intention - i thought it makes sense to "remove" guest cart right after its clear we have a customer cart.
Otherwise it might take n API calls time to delete it - do you see a drawback?

Reply:
Ah okay if its intended that's fine. I just wasn't sure if we should keep the guest cart intact in case of an later error.